### PR TITLE
Flares burn weeds

### PR DESCRIPTION
**Pull request by psykzz**
## About The Pull Request

Gives flares the ability to burn weed nodes under them, but will waste the flare.

## Why It's Good For The Game

Humans have some counter to the regrowth of weeds near their FOB.

## Changelog
:cl:
balance: Flares now instantly burn out weed nodes under them, but will consume all of the fuel in the flare.
/:cl:

### CODE REVIEW

**Comment by Rohesie** [comment]
Add to the changelog that it will consume fire from the flare.
Though won't the weed keep regrowing if it's not a node, drastically reducing the flare life?

**Comment by psykzz** [comment]
done and yes it would, im curious of the impact.

**Comment by Rohesie** [comment]
Let's test it out and run the experiment.

**Comment by DominikPanic** [comment]
@Rohesie Are you happy with the current implementation?

**Comment by Rohesie** [comment]
The idea is good, but this is a flare nerf as it kills its burn time and the weed keeps respawning.
Make it burn nodes if you want, but leave the regular weeds out to avoid the loop.

**Comment by psykzz** [comment]
@Rohesie can i convince you otherwise for flares?
Its a flare nerf yes, but flares are not weak in the first place, if marines are just spamming them about and not clearing nodes they should be penalized for it.

They can easily use it to spread out to an area and regain "control" by removing the node / weeds.

**Comment by psykzz** [comment]
After a long discussion this needs changing to only burn out nodes and consume some fuel from the flare (@Rohesie can you confirm)?

Changes required.

**Comment by Rohesie** [comment]
Yes, that'd be the way to go with it and avoid unnecessary loops.

**Comment by psykzz** [comment]
Changes implemented

File: code/game/objects/items/explosives/grenades/marines.dm [hunk 1]
```diff
@@ -362,3 +363,9 @@ proc/flame_radius(radius = 1, turf/T, burn_intensity = 25, burn_duration = 25, b
 		if(launched && CHECK_BITFIELD(resistance_flags, ON_FIRE))
 			var/armor_block = L.run_armor_check(target_zone, "fire")
 			L.apply_damage(rand(throwforce*0.75,throwforce*1.25), BURN, target_zone, armor_block) //Do more damage if launched from a proper launcher and active
+
+	// Flares instantly burn out nodes when thrown at them.
+	var/obj/effect/alien/weeds/node/N = locate() in loc
+	if(N && istype(N))
+		N.fire_act()
+		turn_off()
```
Comment (Rohesie) [inline code/game/objects/items/explosives/grenades/marines.dm]: Will it consume no fuel?

Comment (psykzz) [reply code/game/objects/items/explosives/grenades/marines.dm]: `turn_off()` sets fuel and active to `0` / `FALSE`.

Comment (Rohesie) [reply code/game/objects/items/explosives/grenades/marines.dm]: So it will immediately snuff out the flare. Alright.

**Pull request merged**